tests/core/pyspec/eth2spec/test/helpers/execution_payload.py

```diff
@@ -1,7 +1,136 @@
+        state_root=execution_payload.state_root,
+        receipts_root=execution_payload.receipts_root,
+        logs_bloom=execution_payload.logs_bloom,
+        prev_randao=execution_payload.prev_randao,
+        block_number=execution_payload.block_number,
+        gas_limit=execution_payload.gas_limit,
+        gas_used=execution_payload.gas_used,
+        timestamp=execution_payload.timestamp,
+        extra_data=execution_payload.extra_data,
+        base_fee_per_gas=execution_payload.base_fee_per_gas,
+        block_hash=execution_payload.block_hash,
+        transactions_root=spec.hash_tree_root(execution_payload.transactions)
+    )
+    if is_post_capella(spec):
+        payload_header.withdrawals_root = spec.hash_tree_root(execution_payload.withdrawals)
+    return payload_header
+
+
+# https://eips.ethereum.org/EIPS/eip-2718
+def compute_trie_root_from_indexed_data(data):
+    """
+    Computes the root hash of `patriciaTrie(rlp(Index) => Data)` for a data array.
+    """
+    t = HexaryTrie(db={})
+    for i, obj in enumerate(data):
+        k = encode(i, big_endian_int)
+        t.set(k, obj)
+    return t.root_hash
+
+
+# https://eips.ethereum.org/EIPS/eip-4895
+def compute_el_header_block_hash(spec,
+                                 payload_header,
+                                 transactions_trie_root,
+                                 withdrawals_trie_root=None):
+    """
+    Computes the RLP execution block hash described by an `ExecutionPayloadHeader`.
+    """
+    execution_payload_header_rlp = [
+        # parent_hash
+        (Binary(32, 32), payload_header.parent_hash),
+        # ommers_hash
+        (Binary(32, 32), bytes.fromhex("1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347")),
+        # coinbase
+        (Binary(20, 20), payload_header.fee_recipient),
+        # state_root
+        (Binary(32, 32), payload_header.state_root),
+        # txs_root
+        (Binary(32, 32), transactions_trie_root),
+        # receipts_root
+        (Binary(32, 32), payload_header.receipts_root),
+        # logs_bloom
+        (Binary(256, 256), payload_header.logs_bloom),
+        # difficulty
+        (big_endian_int, 0),
+        # number
+        (big_endian_int, payload_header.block_number),
+        # gas_limit
+        (big_endian_int, payload_header.gas_limit),
+        # gas_used
+        (big_endian_int, payload_header.gas_used),
+        # timestamp
+        (big_endian_int, payload_header.timestamp),
+        # extradata
+        (Binary(0, 32), payload_header.extra_data),
+        # prev_randao
+        (Binary(32, 32), payload_header.prev_randao),
+        # nonce
+        (Binary(8, 8), bytes.fromhex("0000000000000000")),
+        # base_fee_per_gas
+        (big_endian_int, payload_header.base_fee_per_gas),
+    ]
+    if is_post_capella(spec):
+        # withdrawals_root
+        execution_payload_header_rlp.append((Binary(32, 32), withdrawals_trie_root))
+
```

Comment:
Need to add [eip4844 `excess_data_gas` field](https://eips.ethereum.org/EIPS/eip-4844#header-extension).

```python
if is_post_eip4844(spec):
    # excess_data_gas
    execution_payload_header_rlp.append((big_endian_int, payload_header.excess_data_gas))
```